mesonbuild/cmdline.py

```diff
@@ -78,14 +77,14 @@ def write_cmd_line_file(build_dir: str, options: SharedCMDOptions) -> None:
     filename = get_cmd_line_file(build_dir)
     config = CmdLineFileParser()
 
-    properties: OrderedDict[str, str] = OrderedDict()
+    properties = {}
```

Comment:
This will default to `Dict[Any, Any]`, so `properties: T.Dict[str, T.List[str]]`